### DISCUSSION TITLE
Server error

### DISCUSSION THREAD

**Discussion by muhammedctgr** [nextauthjs/next-auth#5414] [Help]
### Environment

"next":  "^12.0.10"
"react": "17.0.2"
"next-auth":  "^4.10.3"

Windows

Node v16.13.2

### Reproduction URL

github.com/muhammedctgr/p-hammed-blogr-nextjs-prisma

### Describe the issue

I am trying to activate the next-auth with my github. Everything is configured fine, when I connected it with local host and configured the oauth app on github, i can use my github to login to the app, but when I try to connect the nextauth_url with the url of my prospective vercel app...its saying server error.
![Screenshot 2022-09-20 184435](https://user-images.githubusercontent.com/73699892/191328156-fb21cf76-117a-45b1-985a-370b23618aae.jpg)
I've been on this for far too long...any help pls

### How to reproduce

Click on login on the homepage.. It will show sign in with github, after clicking comes this error

### Expected behavior

you should be able to login with your github account

**Comment by balazsorban44** [comment]
Most likely, you forgot to define `SECRET`, based on your code. Remember, on Vercel you don't need to set `NEXTAUTH_URL` as we detect the environment. See: https://next-auth.js.org/deployment#vercel

**Reply by muhammedctgr** [reply]
how do i define the secret, on the repo? and how do i activate/verify logins with github when github is the provider for NEXTAUTH_URL?

**Comment by muhammedctgr** [comment]
this is the error from the logs on vercel...how do i define secret in production?
![Screenshot 2022-09-25 122021](https://user-images.githubusercontent.com/73699892/192140897-b026b86a-657c-4ae8-bafc-49579f17fb5f.jpg)
